composer.json

```diff
@@ -18,10 +18,17 @@
         "ext-ctype": "*",
         "ext-mbstring": "*",
         "ext-pdo_pgsql": "*",
+        "aws/aws-sdk-php-symfony": "^2.0",
+        "csa/guzzle-bundle": "^3.1",
         "doctrine/doctrine-bundle": "^1.10",
         "doctrine/doctrine-migrations-bundle": "^2.0",
+        "guzzlehttp/guzzle": "^6.3",
+        "guzzlehttp/psr7": "^1.5",
+        "league/flysystem": "^1.0",
+        "league/flysystem-aws-s3-v3": "^1.0",
```

Comment:
There are two abstraction layers in play, Flysystem and the `minio/minio` Fake implementation for S3. The first would allow other cloud providers and backends, would it allow a local filesystem option that can substitute S3 for development? Or there is something specific that Flysystem allows on S3, because in that case it would defeat the purpose of Flysystem to abstract it away and allow swapping in e.g. Google Cloud Storage.

Reply:
The code will be in a separate library. I'd like that not to depend on a specific implementation, so Flysystem acts like Doctrine DBAL does for the database. This application has to be opinionated, so will be set up for S3. If you need to use GCS, MySQL or whatever you can have your own application (=config/dependencies) using the same code.